streamer/src/streamer.rs

```diff
@@ -113,13 +122,111 @@ pub fn receiver(
         .unwrap()
 }
 
+#[derive(Debug, Default)]
+struct SendStats {
+    bytes: u64,
+    count: u64,
+}
+
+#[derive(Default)]
+struct StreamerSendStats {
+    host_map: HashMap<[u16; 8], SendStats>,
+    since: Option<Instant>,
+}
+
+impl StreamerSendStats {
+    fn maybe_submit(&mut self, name: &'static str) {
+        const SUBMIT_CADENCE: Duration = Duration::from_secs(10);
+        const MAX_REPORT_ENTRIES: usize = 5;
+        let elapsed = self.since.as_ref().map(Instant::elapsed);
+        if elapsed.map(|e| e < SUBMIT_CADENCE).unwrap_or_default() {
+            return;
+        }
```

Comment:
this probably also needs to submit stats if for example, ```host_map.len() > 1000```, or something, just so that the hashmap does not grow too large.